gui/backend/upsert_bot_config.go

```diff
@@ -178,6 +188,15 @@ func (s *APIServer) validateConfigs(req upsertBotConfigRequest) *upsertBotConfig
 	return nil
 }
 
+func isBotNameValid(botName string) (bool, error) {
+	re, err := regexp.Compile(`^[a-zA-Z0-9\- ]+$`)
```

Comment:
Compiling the regex on startup, rather than per-request, makes sense. But I don't see any existing function called `Init()` - are you suggesting creating this function? I'm implementing the rest of this logic in `cmd/server_amd64.go`, as requested.

Reply:
Also, I'm not sure what "this file" refers to. I'm assuming you mean `gui/backend/upsert_bot_config.go`, because storing this variable within `cmd/server_amd64.go` as a global in the `cmd` package would create an import cycle.